File: solutions/0228-summary-ranges/solution.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> ans = new ArrayList<>();
        int n = nums.length;
        int i = 0;
        int start;
        while(i < n)
        {
            start = i;
            while(i < n-1 && nums[i] + 1 == nums[i+1])
                i++;
            if(i == start)
                ans.add(nums[start]+"");
            else
                ans.add(nums[start]+"->"+nums[i]);
            i++;
        }
        return ans;
    }
}

File: solutions/0228-summary-ranges/solution.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        ans = []
        n = len(nums)
        i = 0
        while(i < n):
            start = i
            while(i < n-1 and nums[i] + 1 == nums[i+1]):
                i += 1
            if(start == i):
                ans.append(str(nums[start]))
            else:
                ans.append(str(nums[start])+"->"+str(nums[i]))
            i += 1

        return ans
